### ISSUE TITLE
UNABLE_TO_VERIFY_LEAF_SIGNATURE error when running commands/authenticating

### ISSUE THREAD

**Issue by raags** [Azure/azure-xplat-cli#3213]
CLI Version: **0.9.8**
OS Type: Linux (Ubuntu)
Installation via:  npm

Mode: **ARM**

Environment: **AzureCloud**

Description:
Any command fails with the following, including `azure login`

```
UNABLE_TO_VERIFY_LEAF_SIGNATURE
Error information has been recorded to /hs/genssh/.azure/azure.err
```

Steps to reproduce:
Run `azure group list`

Error stack trace:

```javascript
{ [Error: UNABLE_TO_VERIFY_LEAF_SIGNATURE]
  stack: [Getter/Setter],
  __frame:
   { name: '__1',
     line: 42,
     file: '/opt/azure-xplat-cli/node_modules/azure-cli/lib/commands/login._js',
     prev: undefined,
     calls: 2,
     active: false,
     offset: 48,
     col: 43 },
  rawStack: [Getter] }
Error: UNABLE_TO_VERIFY_LEAF_SIGNATURE
  <<< async stack >>>
  at __1 (/opt/azure-xplat-cli/node_modules/azure-cli/lib/commands/login._js:90:44)
  <<< raw stack >>>
    at SecurePair.<anonymous> (tls.js:1381:32)
    at SecurePair.emit (events.js:92:17)
    at SecurePair.maybeInitFinished (tls.js:980:10)
    at CleartextStream.read [as _read] (tls.js:472:13)
    at CleartextStream.Readable.read (_stream_readable.js:341:10)
    at EncryptedStream.write [as _write] (tls.js:369:25)
    at doWrite (_stream_writable.js:226:10)
    at writeOrBuffer (_stream_writable.js:216:5)
    at EncryptedStream.Writable.write (_stream_writable.js:183:11)
    at write (_stream_readable.js:602:24)
```

**Comment by raags** [comment]
Using `export NODE_TLS_REJECT_UNAUTHORIZED=0` and then running `azure login` works

**Comment by amarzavery** [comment]
what version of node.js (node -v) is installed on the system?

**Comment by sigurdvh** [comment]
I can repro the same bug under Mac OS X 10.11.6, although I am using the docker container under microsoft/azure-cli to run the cli tools. The node version for the container is v4.2.4

**Comment by yugangw-msft** [comment]
@amarzavery, please open an ICM ticket to ARM team. This is serious.  It will hit any arm client, including new azure-cli.
//cc:@mayurid, @vivsriaus, @ravbhatnagar, @yangl900, @lmazuel 
I think the culprit is the server certificate chain of `westus.management.azure.com` is incomplete. 
Just now ran the [SSLlabs report](https://www.ssllabs.com/ssltest/analyze.html?viaform=on&d=westus.management.azure.com), which confirms this. For easier cross check, this starts to surface on `Friday night (pacific time)`

![image](https://cloud.githubusercontent.com/assets/6014209/19015799/520f2634-87c0-11e6-8be4-2cdceac0d8bd.png)

So in short, this is not specific to azure-cli, any client will hit this. 
```js
var request = require('request');
request('https://management.azure.com', function (error, response, body) {
  if (!error && response.statusCode == 200) {
    console.log(body)
  } else if (error) {
    throw error;
  } else {
    console.log('we are good')
  }
})
```

**Comment by vivsriaus** [comment]
@yugangw-msft @amarzavery @mayurid We've pulled out westus ARM from traffic manager, so the clients shouldn't have any issues. Can you please confirm?

**Comment by amarzavery** [comment]
```
Amars-MacBook-Pro:azure-xplat-cli amarz$ node test.js 
we are good
```

verified it from my home network and it looks good. Thanks for resolving this.

**Comment by yugangw-msft** [comment]
Verified too. :-)
